CMakeLists.txt

```diff
@@ -294,6 +295,42 @@ else()
     message( STATUS "${red}No HIP/ROCm support: gpu_backend = ${gpu_backend}${plain}" )
 endif()
 
+#-------------------------------------------------------------------------------
+# OneMKL/SYCL support.
+message( "" )
+set( blaspp_use_onemkl false )  # output in blasppConfig.cmake.in
+if (NOT gpu_backend MATCHES "^(cuda|hip)$"
+    AND gpu_backend MATCHES "^(onemkl|auto)$" )
+
+    message( STATUS "${bold}Looking for OneMKL${not_bold} (gpu_backend = ${gpu_backend})" )
+    if (gpu_backend STREQUAL "onemkl")
+        find_package( MKL CONFIG REQUIRED HINTS "$ENV{MKL_ROOT}")
+    else()
+        find_package( MKL CONFIG QUIET HINTS "$ENV{MKL_ROOT}")
+    endif()
+    # message(STATUS "Available targets: ${MKL_IMPORTED_TARGETS}")
+
+    if (MKL_FOUND)
+        set( gpu_backend "onemkl" )
+        set( blaspp_defs_onemkl_ "-DBLAS_HAVE_ONEMKL" )
+        set( blaspp_use_onemkl true )
+        # CMake FindBLAS works, SLATE internal BLAS finder may cause problems
+        if (NOT BLA_VENDOR)
+           set( BLA_VENDOR Intel10_64lp )
+        endif()
```

Comment:
What's the issue with the internal BLAS++ finder? Because this hard-codes one specific version of MKL (multi-threaded lp64), ignoring any settings for integer size, threading, etc. that the user may have set.

Reply:
SLATE's internal `BLASFinder.cmake `is not setup for the `icpx` compiler so it does not probe properly.
I will take at `BLASFinder.cmake` to see about integrating `icpx.`